akka-http2-support/src/main/scala/akka/http/impl/engine/http2/BufferedOutletSupport.scala

```diff
@@ -29,15 +29,17 @@ trait BufferedOutletSupport { logic: GraphStageLogic ⇒
     val buffer: java.util.ArrayDeque[T] = new java.util.ArrayDeque[T]
     var pulled: Boolean = false
 
+    def doPush(elem: T): Unit = outlet.push(elem)
```

Comment:
Not sure if you mean another `doPush` here. This one is just an internal hook in `BufferedOutletSupport` to allow subcalsses to override or hook into the pushing mechanic. I guess that's not a final solution but more of a way to get things done right now.